core/lib/dal/src/events_web3_dal.rs

```diff
@@ -97,21 +106,56 @@ impl EventsWeb3Dal<'_, '_> {
                 let topics: Vec<_> = topics.iter().map(H256::as_bytes).collect();
                 query = query.bind(topics);
             }
+
             query = query.bind(limit as i32);
 
-            let db_logs: Vec<StorageWeb3Log> = query
+            let mut db_logs: Vec<StorageWeb3Log> = query
                 .instrument("get_logs")
                 .report_latency()
                 .with_arg("filter", &filter)
                 .with_arg("limit", &limit)
                 .fetch_all(self.storage.conn())
                 .await?;
+
+            if api_mode == APIMode::Modern {
+                db_logs.retain(|log| {
+                    log.address != L2_ETH_TOKEN_ADDRESS.as_bytes()
+                        || log.topic1 != TRANSFER_EVENT_TOPIC.as_bytes()
+                });
+
+                if !db_logs.is_empty() {
+                    let mut new_event_index_in_block = 0;
+                    let mut new_event_index_in_tx = 0;
+                    let mut current_block = db_logs[0].miniblock_number;
+                    let mut current_tx_hash = db_logs[0].tx_hash.clone();
+
+                    for i in 0..db_logs.len() {
```

Comment:
Nit: It is more idiomatic to iterate over mutable reference to logs: `for db_log in &mut db_logs`.